### DISCUSSION TITLE
Modified request URL after upgrading to V1.4.0.

### DISCUSSION THREAD

**Discussion by TomppaPackage** [axios/axios#5880] [Q&A] [answered]
### Describe the issue

After updating from 0.20 to 1.4.0, a request URL query-string serialisation has changed.

Previously, we had this;
```
&sortOptions=%7B%22column%22:%22created_at%22,%22direction%22:%22DESC%22%7D
```

After upgrading to V1, it looks like this
```
&sortOptions[column]=created_at&sortOptions[direction]=DESC
```

Is there a configuration option to use the prior behaviour? I cannot find anything besides `paramsSerializer`, but that doesn't seem to be making any difference.

### Axios Version

1.4.0

**Comment by Vertganti** [comment, marked as answer]
You can use a custom encoder function for this. I wrote the following function which is working for our use case (no guarantees that it will cover every case):

```
function legacyParamsSeralizer (params) {
  let legacyString = ''
  let prefix = ''
  for(let paramKey in params) {
    let paramValue = params[paramKey]
    if (typeof paramValue === 'object') {
      paramValue = JSON.stringify(paramValue)
    }
    legacyString += prefix + encodeURIComponent(paramKey) + '=' + encodeURIComponent(paramValue)
    prefix = '&'
  }
  return legacyString
}
```

You can use the custom encoder for specific calls (example in [this comment](https://github.com/axios/axios/issues/5058#issuecomment-1272107602)), create a custom axios instance using it (example in [this issue](https://github.com/axios/axios/issues/6262#issue-2155843935) or set it as the default encoder function (example in [this comment](https://github.com/axios/axios/issues/1443#issuecomment-1706879463)). I set it as the default encoder function like this:

```
axios.defaults.paramsSerializer = (params) => legacyParamsSeralizer(params)
```
